src/functions/raisebox.js

```diff
@@ -1,20 +1,20 @@
 // @flow
-import defineFunction, {ordargument} from "../defineFunction";
+import defineFunction from "../defineFunction";
 import buildCommon from "../buildCommon";
 import mathMLTree from "../mathMLTree";
 import {assertNodeType} from "../parseNode";
 import {calculateSize} from "../units";
 
+import * as html from "../buildHTML";
 import * as mml from "../buildMathML";
-import * as sizing from "./sizing";
 
 // Box manipulation
 defineFunction({
     type: "raisebox",
     names: ["\\raisebox"],
     props: {
         numArgs: 2,
-        argTypes: ["size", "text"],
+        argTypes: ["size", "hbox"],
```

Comment:
It feels a little weird to introduce a new arg type just to wrap the arg in `\textstyle\textrm{...}`.  Would it be possible to use macro for this?

Reply:
In LaTeX, `\text` is a macro that enters `\hbox` mode, then attempts to restore sizing and font to match the current values in the surrounding math mode.  So it feels like there are fundamentally two different ways to enter text mode: `\text` which preserves everything and `\hbox` which resets everything.  A lot of AMS macros (such as `\text`) behave like the former, while a lot of TeX and LaTeX macros (such as `\raisebox`) behave like the latter.  Hence the second arg type.

I just reviewed all the functions we currently support for those that use `\hbox` behavior.  Here are the ones I could find:

1. `\verb` already has code to deal with this, which could be simplified with the new `hbox` argument type (though is admittedly pretty simple):

   ```js
           // \verb enters text mode and therefore is sized like \textstyle
           const newOptions = options.havingStyle(options.style.text());
   ```

2. `\fbox` does *not* correctly use hbox style; a change from argument type `text` to `hbox` would fix this bug.  For example, `a_{xyz\fbox{xyz}\boxed{xyz}}` currently renders as 
![katex.org rendering](https://user-images.githubusercontent.com/2218736/61187618-bae27880-a641-11e9-8641-4f4dac512398.png) but should render as 
![QuickLaTeX rendering](https://user-images.githubusercontent.com/2218736/61187612-9eded700-a641-11e9-8fef-30645892ecbc.png) (and does so with the `hbox` fix).

3. Other macros that we might one day support that would benefit include `\hbox`, `\makebox`, `\mbox`, `\settoheight`.  I'm not sure how important these are, as a lot of other box macros (e.g. `\settowidth`, `\setbox`/`\box`/`\copy`) would be difficult for us to implement at present.

So I'd suggest either changing `\verb` to use the new functionality, or changing `\raisebox` and `\fbox` to match `\verb`.  I have a vague feeling that using a separate arg type would make it easier to avoid bugs (forces you to make a decision).

Reply:
Another argument for keeping this as is is that it's helpful to have this reset code centralized.  For example, I just found some slight discrepancies from LaTeX:

CORRECT: `a_{\bf xyz\fbox{xyz}\raisebox{0pt}{xyz}\boxed{xyz}}` renders with this PR as
![KaTeX render](https://user-images.githubusercontent.com/2218736/61187816-67255e80-a644-11e9-8e39-734ab90bcf25.png) in KaTeX vs. ![QuickLaTeX render](https://user-images.githubusercontent.com/2218736/61187812-4ceb8080-a644-11e9-8573-b61bc8ca7d21.png) in LaTeX.

INCORRECT: `a_{\textbf{xyz\fbox{xyz}\raisebox{0pt}{xyz}\boxed{xyz}}}` renders with this PR as
![KaTeX render](https://user-images.githubusercontent.com/2218736/61187852-e6b32d80-a644-11e9-9b89-48904775f421.png) in KaTeX vs. ![QuickLaTeX render](https://user-images.githubusercontent.com/2218736/61187839-c97e5f00-a644-11e9-9c66-f9fdd0c09458.png) in LaTeX.

It looks like AMS's `\text` (and `\textbf`) actually redefine the font size of the argument to the script size.  This is a bug for another PR, but I feel like we have a better chance of fixing this across `\fbox` and `\raisebox` (and any other future operations) if they all use common code for shaping an `hbox` argument.

Reply:
I agree that it could be nice to define everything using a new function `\hbox`, though it's a bit tricky currently because macros and functions are a bit annoying to mix. We'd have to define `\fbox` and `\raisebox` as macros that call a function `\hbox` and an auxiliary function like `\@fbox` and `\@raisebox` that do the actual rendering (as in `\def\fbox#1{\@fbox{hbox{#1}}}`).

Perhaps we could refactor `hbox` argument type into a separate `defineFunction` mechanism for preprocessing arguments according to macros, like `argMap: [`\hbox{#1}`]` meaning "for my first argument, please give me `\hbox{#1}` instead of `#1`".  This would be a lot more general, and would keep function definitions very convenient, without having to define macros and intermediate functions like `\@fbox`, and separating out the functionality of `hbox`-style "argument reformattings" from other argument types' "argument parsing".

I think this could be done as a pretty clean PR after this one -- it would be a drop-in replacement for `hbox` argument type, so it's OK if we end up using it more.

Reply:
That's an interesting idea.  Definitely much more flexible that what I was suggesting.  I like it.